AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -11,7 +11,10 @@ protocol MeasurementStreamStorage {
     func createSession(_ session: Session) throws
     func createSessionAndMeasurementStream(_ session: Session, _ stream: MeasurementStream) throws -> MeasurementStreamLocalID
     func updateSessionStatus(_ sessionStatus: SessionStatus, for sessionUUID: SessionUUID) throws
+    func updateSessionEndtime(_ endTime: Date, for sessionUUID: SessionUUID) throws
```

Comment:
This is a throwing one too, I need to ask why. Maybe I am wrong 😅

Reply:
Well, this func has to throw because it calls `context.existingSession(uuid: sessionUUID)` (and the session may not exist) and also it's trying to call `context.save()`